src/Symfony/Component/HttpFoundation/Request.php

```diff
@@ -1705,6 +1710,30 @@ public function isXmlHttpRequest()
         return 'XMLHttpRequest' == $this->headers->get('X-Requested-With');
     }
 
+    /**
+     * Checks whether the client browser prefers safe content or not according to RFC8674.
+     *
+     * @see https://tools.ietf.org/html/rfc8674
+     *
+     * @return bool
+     */
+    public function isSafePreferred(): bool
+    {
+        if (null !== $this->isSafePreferred) {
+            return $this->isSafePreferred;
+        }
+
+        if (!$this->isSecure()) {
+            $this->isSafePreferred = false;
+
+            return $this->isSafePreferred;
+        }
+
+        $this->isSafePreferred = in_array('safe', (AcceptHeader::fromString($this->headers->get('Prefer'))->all()), false);
```

Comment:
I'm not this is okay to use AcceptHeader capability but the [RFC 7240](https://tools.ietf.org/html/rfc7240#section-1.1) defines the behavior of `Prefer` header which seems similar to `Accept-x` ones

Reply:
Unless I'm wrong, tests fail because the `headers->get('Prefer')` only returns the value for the first `Prefer` header. I'm not sure how to (and I should) make the AcceptHeader works with multiple value